docker-compose.yml

```
@@ -1,49 +1,90 @@
 version: '3.7'
 services:
   postgres:
-    image: postgres:9.6
+    image: postgres:12
     environment:
       - POSTGRES_USER=airflow
       - POSTGRES_PASSWORD=airflow
       - POSTGRES_DB=airflow
       - PGDATA=/var/lib/postgresql/data/pgdata
     volumes:
-      - ./pgdata:/var/lib/postgresql/data/pgdata
+      - pgdata:/var/lib/postgresql/data/pgdata
     logging:
       options:
         max-size: 10m
         max-file: "3"
-  webserver:
-    image: geoscienceaustralia/airflow
-    restart: always
+  setup-airflow:
+    image: apache/airflow:1.10.12
     depends_on:
       - postgres
-    environment:
-      - LOAD_EX=n
+    environment: &env
+      - AIRFLOW__CORE__SQL_ALCHEMY_CONN=postgresql://airflow:airflow@postgres/airflow
       # Uncomment and add fernet key to encrypt/decrypt secrets. For details see
       # https://airflow.readthedocs.io/en/stable/howto/secure-connections.html
-      # - FERNET_KEY=<insert-fernet-key-here>
-      - EXECUTOR=Local
+      - AIRFLOW__CORE__FERNET_KEY=${FERNET_KEY}
+      # - EXECUTOR=Local
       - AIRFLOW__SMTP__SMTP_HOST=maildev
       - AIRFLOW__CORE__ENABLE_XCOM_PICKLING=False
+      - AIRFLOW__CORE__EXECUTOR=SequentialExecutor
+      - HOST_USER_ID=1000
+      - HOST_GROUP_ID=1000
+    command: "bash -c 'airflow upgradedb && airflow connections --add --conn_id lpgs_gadi --conn_uri ssh://dra547@gadi.nci.org.au/ && airflow connections --add --conn_id dea_public_data_upload --conn_uri s3://foo:bar@dea-public-data-dev/ && airflow connections --add --conn_id aws_nci_db_backup --conn_uri s3://foo:bar@dea-db-backups/'"
+  webserver:
+    image: apache/airflow:1.10.12
+    restart: always
+    depends_on:
+      - postgres
+      - setup-airflow
+    environment: *env
     logging:
       options:
         max-size: 10m
         max-file: "3"
     volumes:
-      - ./dags:/usr/local/airflow/dags
-      - ./plugins:/usr/local/airflow/plugins
-      - ./scripts:/usr/local/airflow/scripts
-      - ./requirements.txt:/requirements.txt
+      - ./dags:/opt/airflow/dags
+      - ./plugins:/opt/airflow/plugins
+      - ./scripts:/opt/airflow/scripts
+      - logs/:/opt/airflow/logs
+      - ./requirements.txt:/opt/airflow/requirements.txt
+      - /dev/urandom:/dev/random   # Required to get non-blocking entropy source
     ports:
       - "8080:8080"
     command: webserver
     healthcheck:
-      test: ["CMD-SHELL", "[ -f /usr/local/airflow/airflow-webserver.pid ]"]
+      test: ["CMD-SHELL", "[ -f /opt/airflow/airflow-webserver.pid ]"]
+      interval: 30s
+      timeout: 30s
+      retries: 3
+  scheduler:
+    image: apache/airflow:1.10.12
+    restart: always
+    depends_on:
+      - postgres
+      - setup-airflow
+    environment: *env
+    logging:
+      options:
+        max-size: 10m
+        max-file: "3"
+    volumes:
+      - ./dags:/opt/airflow/dags
+      - ./plugins:/opt/airflow/plugins
+      - ./scripts:/opt/airflow/scripts
+      - logs/:/opt/airflow/logs
```

```suggestion
      - ./logs/:/opt/airflow/logs
```
And make the logs dir with a .gitkeep would work, right?